share/scripts/sdcdb.tcl

```diff
@@ -515,13 +517,14 @@ proc list_pc {{x0 -1} {x1 9}} {
 proc list_address {address {x0 -1} {x1 9} {showerror 0}} {
     variable addr2file
     set address [expr {$address}]
-    if {![info exists addr2file($address)]} {
+    set record [lindex [array get addr2file $address] 1]
+    if {$record eq {}} {
         if {$showerror} {
             error "database address not found for 0x[h $address]"
         }
-        return [puts "no file"]
+        return "no file for address 0x[h $address]"
```

Comment:
Detail: currently the string is different for the `showerror` = `false` vs `true` case. Or maybe that's intentional?
Would it be better to calculate the common message into a variable, and then either `error` or `return` that string?